lib/Diagnostics/Diagnostics.js

```diff
@@ -44,12 +128,134 @@ class Diagnostics {
+                    alerts: [...this.reportedAlerts]
+                };
+                return BackendClient.sendBeacon(this.entryId, this.partnerId, data, 'healthData');
+            });
+        });
+    }
+
+    _runDiagnosticsReport() {
+        if (!this._shouldRun) {
+            this._logger.debug('Diagnostics module stopped. Exiting runDiagnosticsReport method');
+            this._diagnosticsEnd.resolve();
+            return Q.resolve();
+        }
+
+        return Q.fcall(() => {
+            // If staticInfo is modified -> update server beacon
+            this._logger.debug('Starting Diagnostic report process');
+            if (this.staticMetaData.modified) {
+                this._logger.info('Entry static information has been modified. Reporting [staticData] beacon');
+                return BackendClient.sendBeacon(this.entryId, this.partnerId, this.staticMetaData.data, 'staticData')
+                    .then(()=> {
+                        this.staticMetaData.modified = false;
+                    });
+            }
+        })
+        .then(() => {
+            // Update dynamic info in every iteration
+            if (this.dynamicMetaData.flavors.length) {
+                this._logger.info('Sending [dynamicData] beacon');
+                return BackendClient.sendBeacon(this.entryId, this.partnerId, this.dynamicMetaData, 'dynamicData');
+            }
+        })
+        .then(() => {
+            // Clean alerts that are past the window size
+            this._cleanAlerts();
+        })
+        .finally(() => {
+            setTimeout(this._runDiagnosticsReport.bind(this), this._reportIntervalTimeInMSec);
+            if (!this._shouldRun) {
+                this._diagnosticsEnd.resolve();
+            }
+        });
+
+    }
+
+    handleDiagnosticsReport(metaData) {
+        // Check should be performed only when diagnostics object is updated
+        if (this.lastDiagnosticsObjTime === metaData.source.currentTime)
+            return;
+
+        this._diagnosticsAnalyzer.runFullDiagnosticsAnalysis(metaData.source);
+        this._setStaticMetaData(metaData);
+        this._setDynamicMetaData(metaData);
+
+        this.lastDiagnosticsObjTime = metaData.source.currentTime;
+    }
+
+    _setStaticMetaData(metaData) {
+        // There are 3 scenarios where we would send the static data to server:
+        // (1) If Diagnostics first receives metaData -> send beacon
+        // (2) If one of the inputs was removed or another has been added -> send beacon
+        // (3) If an input was changed and now has a different sessionId -> send beacon
+        let tempStaticData = JSON.parse(JSON.stringify({ 'inputs': metaData.source.inputs, 'playlistGenerator': metaData.playlistGenerator }));
+        _.each(tempStaticData.inputs, i => {
+           delete i.IOPerformance;
+           delete i.Encoder.syncPTSData;
+           delete i.Properties.timeRunningSeconds;
+        });
+
+        if (!this.staticMetaData.data) {
```

Comment:
you can do it with one line: this.staticMetaData.modified = !_.isEqual(tempStaticData, this.staticMetaData.data)